### PR TITLE
ddl: validate table info before doing ddl job to avoid load infoschema error

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Validate table info before doing ddl job to avoid load infoschema error.
eg1: 
```sql
# Below SQL should return an error when creating table, but in old version tidb will return success, and then load schema will error. 
CREATE TABLE t1 (
	c0 int(11) ,
  	c1 int(11),
    c2 decimal(16,4) GENERATED ALWAYS AS ((case when (c0 = 0) then 0 when (c0 > 0) then (c1 / c0) end))
);
```

eg2:
```sql
create table t1 (a int,b int) partition by hash(a) partitions 4;
# Below SQL should return an error when alter table, but in old version tidb will return success, and then load schema will error. 
 alter table t1 drop column a;
```

### What is changed and how it works?
Add check table info in create table and drop column. 
Maybe add this check in other public path is better. 

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Possible performance regression
Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -242,6 +242,11 @@ func onDropColumn(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		colInfo.State = model.StateWriteOnly
 		// Set this column's offset to the last and reset all following columns' offsets.
 		adjustColumnInfoInDropColumn(tblInfo, colInfo.Offset)
+		err = checkTableInfoValid(tblInfo)
```
Comment (winkyao) [inline ddl/column.go]: we should check if the tableInfo valid before we overwrite the tableInfo in TiKV.

Comment (crazycs520) [reply ddl/column.go]: done. PTAL

Comment (zimulala) [reply ddl/column.go]: We have added `check` on line 265, why do it again?

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -1114,6 +1114,11 @@ func (d *ddl) CreateTableWithLike(ctx sessionctx.Context, ident, referIdent ast.
 	return errors.Trace(err)
 }
 
+func checkTableInfoValid(tblInfo *model.TableInfo) error {
```
Comment (zimulala) [inline ddl/ddl_api.go]: Do we need to add comments for it?

**Pull request merged**